rackcity/models/asset.py

```diff
@@ -217,6 +217,7 @@ class Meta:
 
     def save(self, *args, **kwargs):
         try:
+            self.full_clean()
```

Comment:
what is this?

Reply:
`full_clean()` is essentially the validation method on a model -- it cleans the data for the database and will throw a ValidationError if any validation step fails.  We thought that it was called automatically on both serializer.save() and model.save(), but turns out it's only called on the serializer.  Read more about what it does here: https://docs.djangoproject.com/en/3.0/ref/models/instances/#validating-objects

Reply:
HOW FUN!